### PR TITLE
Least-effort aliases

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: core/experiment.js [hunk 1]
```diff
@@ -193,14 +194,22 @@ function buildstageList(graphData, tags, require) {
...
-    var streams = linear[i].map(function(pipe, idx) {
+    var streams = [];
+    linear[i].forEach(function(pipe, idx) {
       if (pipe.stageName == undefined)
         pipe.stageName = 'passthrough';
       var thisStream = stageLoader.stageSpecificationToStage({name: pipe.stageName, options: pipe.options});
-      thisStream.setInput('efrom', pipe.id);
-      thisStream.setOutput('eto', pipe.id);
-      return thisStream;
+      if (thisStream.__proto__ == [].__proto__) {
```
Comment (dstockwell) [inline core/experiment.js]: is this just `thisStream instanceof Array` ?

File: core/phase-lib.js [hunk 2]
```diff
@@ -183,7 +184,7 @@ module.exports.fileToBuffer = phase({input: types.string, output: types.buffer,
     if (!tags.filename) {
       this.tags.tag('filename', filename);
     }
-    return fs.readFileAsync(filename);
+    return fs.readFileAsync(filename).then(function(data) { return data; });
```
Comment (dstockwell) [inline core/phase-lib.js]: wha?

File: core/phase.js [hunk 3]
```diff
@@ -349,6 +349,7 @@ function putFunction(type) {
     // TODO: This misses tags when they are set after calling put().
     flowItemPut(this, this.tags.tags);
     this.tags.tag(type.key, type.value);
+    // delete this.tags.tags['from'];
```
Comment (dstockwell) [inline core/phase.js]: ?

File: exports.js [hunk 4]
```diff
@@ -12,7 +12,7 @@
 */
 
 'use strict';
-module.exports.phase = require('./core/phase-register');
+module.exports.phase = require('./core/register').phase;
```
Comment (dstockwell) [inline exports.js]: export alias here too

**Comment by dstockwell** [comment]
lgtm

File: core/experiment.js (hunk 1 above)
Comment (shans) [inline core/experiment.js]: ...maybe (done)

File: core/phase-lib.js (hunk 2 above)
Comment (shans) [inline core/phase-lib.js]: Good question.

File: core/phase.js (hunk 3 above)
Comment (shans) [inline core/phase.js]: Removed.

File: exports.js (hunk 4 above)
Comment (shans) [inline exports.js]: done.

**Pull request merged**